### PR TITLE
Adding Image::ToVector and Image::ToScalar

### PR DESCRIPTION
**Pull request by blowekamp**
Add new method to efficiently convert between scalar and vector images, where the vector component is converted to a image dimension. This enable conversion of images when the fastest axis is no used for channels and other operations dones on a per-channel basis.

### CODE REVIEW

File: CMake/sitkUseGTest.cmake [hunk 1]
```diff
@@ -18,7 +18,7 @@ include(GoogleTest)
 
 if (TARGET GTest::Main AND TARGET GTest::GTest)
     message(STATUS "GTest targets already defined.")
-#    return()
+    return()
```
Comment (blowekamp) [inline CMake/sitkUseGTest.cmake:21]: This change should not be part of this PR.

File: Code/Common/include/sitkImage.h [hunk 2]
```diff
@@ -371,6 +371,38 @@ namespace simple
 
     std::string ToString( ) const;
 
+    /** \brief Convert the a first dimension to the components for image with vector pixel type.
```
Comment (zivy) [inline Code/Common/include/sitkImage.h:374]: "the a" -> "the"

File: Code/Common/include/sitkImageConvert.h [hunk 3]
```diff
@@ -31,7 +31,9 @@ namespace simple
 {
 
 
-/** \brief A utility method to help convert between itk image types efficiently.
+/** \brief A utility methods to  convert between itk image types efficiently by
```
Comment (zivy) [inline Code/Common/include/sitkImageConvert.h:34]: "methods" -> "method"

File: Code/Common/include/sitkImage.h [hunk 4]
```diff
@@ -371,6 +371,38 @@ namespace simple
...
+     *
+     * This method will convert a scalar image to a vector image with
+     * the number of components equal to the size of the first
+     * dimension. If the image is already a vector image then the
+     * image is returned.
```
Comment (zivy) [inline Code/Common/include/sitkImage.h:379]: Should it also say something about loosing the metadata information for the first axis? What I mean is that the operation is lossy with respect to metadata: `scalarImage.ToVectorImage().ToScalarImage() + scalarImage` will throw an exception because the metadata is different.

**Review by zivy — APPROVED** [review]
LGTM. Some minor comments with respect to documentation.

File: Code/Common/include/sitkImage.h (hunk 4 above)
Comment (blowekamp) [reply Code/Common/include/sitkImage.h:379]: This is a more complicated issues than just documentation. The spacing and origin have a straight forward way to drop  a dimension and prepend a 1 or 0 respectfully when adding a dimensions. However the Direction matrix can have some degenerative cases when the extract sub-matrix is not appropriate.

The most similar thing that exists in ITK is the ExtractImageFilter, which has multiple direction collapse strategies and not working default.

Comment (zivy) [reply Code/Common/include/sitkImage.h:379]: A possible punting approach, which I think is valid, is to limit these operations to images with identity direction cosine matrices. I suspect this will address all relevant use cases. 

Extend the capabilities of these functions to non identity cosine matrices only if the need arises. Add the functionality only when there is a concrete use case.

Comment (blowekamp) [reply Code/Common/include/sitkImage.h:379]: >A possible punting approach, which I think is valid, is to limit these operations to images with identity direction cosine matrices. I suspect this will address all relevant use cases.

Perhaps just the "component" dimension needs to be identity. This would follow the intended functionality of converting to and from vector images.

**Review by zivy — APPROVED** [review]
